### PR TITLE
Add serialization for isDeleted

### PR DESCRIPTION
**Pull request by grouma**
Also add missing copyright.

### CODE REVIEW

File: build_runner/lib/src/asset/finalized_reader.dart [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
```
Comment (natebosch) [inline build_runner/lib/src/asset/finalized_reader.dart]: [nit] extra blank line below the copyright to separate it from imports

File: build_runner/test/common/matchers.dart [hunk 2]
```diff
@@ -30,6 +30,10 @@ class _AssetGraphMatcher extends Matcher {
     if (graph.allNodes.length != _expected.allNodes.length) matches = false;
     for (var node in graph.allNodes) {
       var expectedNode = _expected.get(node.id);
+      if (node.isDeleted != node.isDeleted) {
```
Comment (jakemac53) [inline build_runner/test/common/matchers.dart]: one of these should be `expectedNode.isDeleted` ;)

File: build_runner/test/common/matchers.dart [hunk 3]
```diff
@@ -30,6 +30,10 @@ class _AssetGraphMatcher extends Matcher {
...
+        matchState['IsDeleted'] = [node.isDeleted, expectedNode.isDeleted];
```
Comment (jakemac53) [inline build_runner/test/common/matchers.dart]: the string here is the description and should include the asset id, something like `IsDeleted of ${node.id}`

File: build_runner/lib/src/asset/finalized_reader.dart (hunk 1 above)
Comment (grouma) [reply build_runner/lib/src/asset/finalized_reader.dart]: Done here and elsewhere.

File: build_runner/test/common/matchers.dart (hunk 3 above)
Comment (grouma) [reply build_runner/test/common/matchers.dart]: Done.

File: build_runner/test/common/matchers.dart (hunk 2 above)
Comment (grouma) [reply build_runner/test/common/matchers.dart]: You don't like tautologies?

**Pull request merged**